awx/main/expect/isolated_manager.py

```diff
@@ -381,10 +381,14 @@ def update_capacity(cls, instance, task_result, awx_application_version):
             logger.error(err_template.format(instance.hostname, instance.version, awx_application_version))
             instance.capacity = 0
         else:
-            if instance.capacity == 0 and task_result['capacity']:
+            if instance.capacity == 0 and task_result['capacity_cpu']:
                 logger.warning('Isolated instance {} has re-joined.'.format(instance.hostname))
-            instance.capacity = int(task_result['capacity'])
-        instance.save(update_fields=['capacity', 'version', 'modified'])
+            instance.cpu_capacity = int(task_result['capacity_cpu'])
+            instance.mem_capacity = int(task_result['capacity_mem'])
+            instance.capacity = get_system_task_capacity(scale=instance.capacity_adjustment,
```

Comment:
Seems like this method could just take an instance. At least we could reuse `instance.cpu_capacity` and `instance.mem_capacity` instead of casting the `task_result` again.

Reply:
Possibly! Part of the reason I separated `get_system_task_capacity()` from the models is to isolate the capacity algorithm itself away from the database, even the fields on the database.

This helps with things like unit testing but also being able to use these values purely for display or debugging purposes.